### ISSUE TITLE
Allow reviews without titles

### ISSUE THREAD

**Issue by WesleyAC** [bookwyrm-social/bookwyrm#2693]
I don't generally want to come up with a snappy title for my review, I just want to write what I think.

To get around the required title, have been writing the name of the book as the title, which looks in the RSS feed something like:

> Review of "Pirate Enlightenment, or the Real Libertalia" (3 stars): Pirate Enlightenment, or the Real Libertalia

But this repetition is annoying to me — instead, I'd like to be able to omit the title, and have the title in RSS simply be:

> Review of "Pirate Enlightenment, or the Real Libertalia" (3 stars)

I think the feed UI would look just fine omitting the title, and having just the stars and text. I'm not sure how this would affect federation, exactly, since I don't know the shape of the non-`Note` AP objects that BookWyrm sends out.

**Comment by j6k4m8** [comment]
+1 to this — I often rate a book without writing a review. Maybe we could auto-generate a title upon submission if one isn't provided, "Jordan rated 'Oryx & Crake' 5 Stars"? Would love the opportunity to get involved in dev, and this seems like a bite-sized effort if you'd be interested in a PR :)

**Comment by JonathanDoughty** [comment]
New BookWrym user here so please forgive if I'm just missing something. A related issue I'm encountering: a few books failed to import successfully from a GoodReads export. Searching turned up entries that I proceeded to add to my read books list with dates. But BookWrym seemingly won't allow me to give the book the same rating as I had long ago without at least giving that rating a title. Related: it was not obvious to me that review text is not required at least in that case.

As a user I'd like to replicate my imported book ratings without trying to dredge up thoughts about books and posting those by now less accurate impressions.

**Comment by hughrun** [comment]
This should be an easy fix since `Name` is not actually required in the model, but for some reason is `required` in the template.

Related to #3234

**Comment by mouse-reeve** [comment]
I think the complicating factor is that they federate as Articles, which (just off the top of my head?) I think do need the title field

**Comment by hughrun** [comment]
hmm you're right. Maybe auto-generation if it's empty is the way to go, per @j6k4m8's suggestion.

**Comment by mouse-reeve** [comment]
Auto generating seems good to me, especially since I’ve been sort of vaguely intending to make them federate as notes, since mastodon handles articles so poorly

**Comment by hughrun** [comment]
> Auto generating seems good to me, especially since I’ve been sort of vaguely intending to make them federate as notes, since mastodon handles articles so poorly

Yeah I was about to reference that (which would allow us to just make the title/name optional) but I'd like to see what would happen if we sent the `type` as `["Note", "Article"]` (see [activitystreams-core](https://www.w3.org/TR/activitystreams-core/#asobject)). If that works in Mastodon, then we'd be able to make it a `Note` for Mastodon and and `Article` for apps that recognise those. But ...I don't know if this will work, so I'll have to do some experiments.
